### PR TITLE
8293188: x86_64: Introduce stubGenerator_x86_64.hpp

### PR DESCRIPTION
**Pull request by iwanowww**
Introduce stubGenerator_x86_64.hpp and extract StubGenerator class declaration there. 

Testing: hs-tier1 - hs-tier4

/cc hotspot

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8293188](https://bugs.openjdk.org/browse/JDK-8293188): x86_64: Introduce stubGenerator_x86_64.hpp

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [93b6ddee](https://git.openjdk.org/jdk/pull/10111/files/93b6ddee538e09b75ed1a91cbfdb31e1b13d2bf8)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10111/head:pull/10111` \
`$ git checkout pull/10111`

Update a local copy of the PR: \
`$ git checkout pull/10111` \
`$ git pull https://git.openjdk.org/jdk pull/10111/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10111`

View PR using the GUI difftool: \
`$ git pr show -t 10111`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10111.diff">https://git.openjdk.org/jdk/pull/10111.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/x86/stubGenerator_x86_64.cpp [hunk 1]
```diff
@@ -61,4889 +48,4921 @@
 #if INCLUDE_JFR
 #include "jfr/support/jfrIntrinsics.hpp"
 #endif
+#include "stubGenerator_x86_64.hpp"
```
Comment (dholmes-ora) [inline src/hotspot/cpu/x86/stubGenerator_x86_64.cpp:51]: This should be in alphabetical order prior to the conditional includes.

File: src/hotspot/cpu/x86/stubGenerator_x86_64.hpp [hunk 2]
```diff
@@ -0,0 +1,450 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
```
Comment (dholmes-ora) [inline src/hotspot/cpu/x86/stubGenerator_x86_64.hpp:2]: As the content of this file has been moved from another, the original file's copyright years should be preserved. Thanks.

File: src/hotspot/cpu/x86/stubGenerator_x86_64.hpp [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
```
Comment (dholmes-ora) [inline src/hotspot/cpu/x86/stubGenerator_x86_64.hpp:24]: Do we need the usual header file include guard?

**Review by dholmes-ora — APPROVED** [review]
Rather tricky to review  but as long as things build okay this should be fine. I'm assuming the motivation here was just to get smaller files to work with?

A couple of minor nits.

Thanks.

File: src/hotspot/cpu/x86/stubGenerator_x86_64.cpp (hunk 1 above)
Comment (iwanowww) [reply src/hotspot/cpu/x86/stubGenerator_x86_64.cpp]: Fixed.

File: src/hotspot/cpu/x86/stubGenerator_x86_64.hpp (hunk 2 above)
Comment (iwanowww) [reply src/hotspot/cpu/x86/stubGenerator_x86_64.hpp]: Fixed.

File: src/hotspot/cpu/x86/stubGenerator_x86_64.hpp (hunk 3 above)
Comment (iwanowww) [reply src/hotspot/cpu/x86/stubGenerator_x86_64.hpp:24]: Good point. Fixed.

Comment (iwanowww) [reply src/hotspot/cpu/x86/stubGenerator_x86_64.hpp:24]: Good point. Fixed.

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by iwanowww** [comment]
Thanks for the reviews, David and Vladimir.

> I'm assuming the motivation here was just to get smaller files to work with?

Yes, I'm looking at splitting `stubGenerator_x86_64.cpp` and grouping stub implementations based on the area they belong to.

**Review by dholmes-ora — APPROVED** [review]

**Comment by iwanowww** [comment]
Thanks for the reviews, David and Vladimir.

/integrate

**Pull request closed without merging**